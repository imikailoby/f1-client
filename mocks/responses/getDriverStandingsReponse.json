{
  "MRData": {
    "xmlns": "http://ergast.com/mrd/1.5",
    "series": "f1",
    "url": "http://ergast.com/api/f1/driverstandings/1.json",
    "limit": "5",
    "offset": "0",
    "total": "72",
    "StandingsTable": {
      "driverStandings": "1",
      "StandingsLists": [
        {
          "season": "1950",
          "round": "7",
          "DriverStandings": [
            {
              "position": "1",
              "positionText": "1",
              "points": "30",
              "wins": "3",
              "Driver": {
                "driverId": "farina",
                "url": "http://en.wikipedia.org/wiki/Nino_Farina",
                "givenName": "Nino",
                "familyName": "Farina",
                "dateOfBirth": "1906-10-30",
                "nationality": "Italian"
              },
              "Constructors": [
                {
                  "constructorId": "alfa",
                  "url": "http://en.wikipedia.org/wiki/Alfa_Romeo_in_Formula_One",
                  "name": "Alfa Romeo",
                  "nationality": "Swiss"
                }
              ]
            }
          ]
        },
        {
          "season": "1951",
          "round": "8",
          "DriverStandings": [
            {
              "position": "1",
              "positionText": "1",
              "points": "31",
              "wins": "3",
              "Driver": {
                "driverId": "fangio",
                "url": "http://en.wikipedia.org/wiki/Juan_Manuel_Fangio",
                "givenName": "Juan",
                "familyName": "Fangio",
                "dateOfBirth": "1911-06-24",
                "nationality": "Argentine"
              },
              "Constructors": [
                {
                  "constructorId": "alfa",
                  "url": "http://en.wikipedia.org/wiki/Alfa_Romeo_in_Formula_One",
                  "name": "Alfa Romeo",
                  "nationality": "Swiss"
                }
              ]
            }
          ]
        },
        {
          "season": "1952",
          "round": "8",
          "DriverStandings": [
            {
              "position": "1",
              "positionText": "1",
              "points": "36",
              "wins": "6",
              "Driver": {
                "driverId": "ascari",
                "url": "http://en.wikipedia.org/wiki/Alberto_Ascari",
                "givenName": "Alberto",
                "familyName": "Ascari",
                "dateOfBirth": "1918-07-13",
                "nationality": "Italian"
              },
              "Constructors": [
                {
                  "constructorId": "ferrari",
                  "url": "http://en.wikipedia.org/wiki/Scuderia_Ferrari",
                  "name": "Ferrari",
                  "nationality": "Italian"
                }
              ]
            }
          ]
        },
        {
          "season": "1953",
          "round": "9",
          "DriverStandings": [
            {
              "position": "1",
              "positionText": "1",
              "points": "34.5",
              "wins": "5",
              "Driver": {
                "driverId": "ascari",
                "url": "http://en.wikipedia.org/wiki/Alberto_Ascari",
                "givenName": "Alberto",
                "familyName": "Ascari",
                "dateOfBirth": "1918-07-13",
                "nationality": "Italian"
              },
              "Constructors": [
                {
                  "constructorId": "ferrari",
                  "url": "http://en.wikipedia.org/wiki/Scuderia_Ferrari",
                  "name": "Ferrari",
                  "nationality": "Italian"
                }
              ]
            }
          ]
        },
        {
          "season": "1954",
          "round": "9",
          "DriverStandings": [
            {
              "position": "1",
              "positionText": "1",
              "points": "42",
              "wins": "6",
              "Driver": {
                "driverId": "fangio",
                "url": "http://en.wikipedia.org/wiki/Juan_Manuel_Fangio",
                "givenName": "Juan",
                "familyName": "Fangio",
                "dateOfBirth": "1911-06-24",
                "nationality": "Argentine"
              },
              "Constructors": [
                {
                  "constructorId": "maserati",
                  "url": "http://en.wikipedia.org/wiki/Maserati",
                  "name": "Maserati",
                  "nationality": "Italian"
                },
                {
                  "constructorId": "mercedes",
                  "url": "http://en.wikipedia.org/wiki/Mercedes-Benz_in_Formula_One",
                  "name": "Mercedes",
                  "nationality": "German"
                }
              ]
            }
          ]
        }
      ]
    }
  }
}
