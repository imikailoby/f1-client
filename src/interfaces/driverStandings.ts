import { Constructor } from './constructor';
import { Driver } from './driver';

export interface DriverStandingsItem {
  Constructors: Constructor[];
  Driver: Driver,
  points: string;
  positionText: string;
  wins: string;
}
