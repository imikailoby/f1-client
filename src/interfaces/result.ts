import { Constructor } from './constructor';
import { Driver } from './driver';

export interface Result {
  number: string;
  position: string;
  positionText: string;
  points: string;
  Driver: Driver;
  Constructor: Constructor;
  grid: string;
  laps: string;
  status: string;
  Time: {
    millis: string;
    time: string;
  };
  FastestLap: {
    rank: string;
    lap: string;
    Time: {
      time: string;
    };
    AverageSpeed: {
      units: string;
      speed: string;
    };
  };
}
