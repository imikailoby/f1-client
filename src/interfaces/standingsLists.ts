import { DriverStandingsItem } from './driverStandings';

export interface StandingsListItem {
  season: string;
  round: string;
  DriverStandings: [DriverStandingsItem];
}
