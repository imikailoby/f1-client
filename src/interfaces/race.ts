import { Result } from './result';

export interface Race {
  season: string;
  round: string;
  url: string;
  raceName: string;
  Circuit: {
    circuitId: string;
    url: string;
    circuitName: string;
    Location: {
      lat: string;
      long: string;
      locality: string;
      country: string;
    };
  };
  date: string;
  time: string;
  Results: [Result];
}
