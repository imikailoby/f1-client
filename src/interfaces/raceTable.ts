import { Race } from './race';

export interface RaceTable {
  season: string;
  position: string;
  Races: Race[];
}
