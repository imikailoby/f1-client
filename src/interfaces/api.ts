import { MRData } from '../types/mrData';
import { RaceTable } from './raceTable';
import { StandingsListItem } from './standingsLists';

export interface IGetDriverStandingsResponse {
  MRData: MRData<{ StandingsTable: { StandingsLists: StandingsListItem[] } }>
}

export interface IGetResultsByYearResponse {
  MRData: MRData<{ RaceTable: RaceTable }>
}

export interface IGetDriverStandings {
  standingsLists: StandingsListItem[],
  pagination: {
    limit: MRData['limit'];
    offset: MRData['offset'];
    total: MRData['total'];
  }
}
