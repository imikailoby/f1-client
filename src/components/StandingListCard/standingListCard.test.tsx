/**
 * @jest-environment jsdom
 */
import { fireEvent, render } from '@testing-library/react';
import getDriverStandingsReponse from '../../../mocks/responses/getDriverStandingsReponse.json';
import { StandingsListItem } from '../../interfaces/standingsLists';
import { StandingListCard } from '.';

const MOCKED_DATA = getDriverStandingsReponse.MRData.StandingsTable.StandingsLists as unknown as StandingsListItem[];

describe('<StandingListCard />', () => {
  it('renders with driver info without crashing', () => {
    const { container } = render(
      <StandingListCard
        onClickCallback={() => undefined}
        totalItems={MOCKED_DATA.length}
        index={0}
        data={MOCKED_DATA}
        style={{ margin: 0 }}
      />,
    );
    expect(container).toMatchSnapshot();
  });
  it('renders with loading state without crashing', () => {
    const { container } = render(
      <StandingListCard
        onClickCallback={() => undefined}
        totalItems={MOCKED_DATA.length + 1}
        index={MOCKED_DATA.length}
        data={MOCKED_DATA}
        style={{ margin: 0 }}
      />,
    );
    expect(container).toMatchSnapshot();
  });
  it('should handle click by content block', () => {
    const onClickCallback = jest.fn();
    const { container } = render(
      <StandingListCard
        onClickCallback={onClickCallback}
        totalItems={MOCKED_DATA.length}
        index={0}
        data={MOCKED_DATA}
        style={{ margin: 0 }}
      />,
    );

    container.getElementsByClassName('content');
    fireEvent.click(container.getElementsByClassName('content')[0]);
    expect(onClickCallback).toHaveBeenCalled();
  });
});
