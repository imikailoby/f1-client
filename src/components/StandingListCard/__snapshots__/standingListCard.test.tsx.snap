// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StandingListCard /> renders with driver info without crashing 1`] = `
<div>
  <div
    class="card "
    style="margin: 0px;"
  >
    <p
      class="label"
    >
      1950
    </p>
    <div
      class="content"
    >
      <div
        class="contentWrapper"
      >
        <div
          class="driver"
        >
          <div
            class="avatar avatar"
          >
            <span>
              ...
            </span>
            <span
              class="number"
            >
              N/A
            </span>
          </div>
          <p
            class="name"
          >
            Nino
             
            <b>
              Farina
            </b>
          </p>
          <p
            class="car"
          >
            Alfa Romeo
          </p>
        </div>
        <p
          class="pts"
        >
          30 PTS
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<StandingListCard /> renders with loading state without crashing 1`] = `
<div>
  <div
    class="card last"
    style="margin: 0px;"
  >
    <p
      class="label"
    >
      ...
    </p>
    <div
      class="content"
    >
      <div
        class="contentWrapper"
      >
        <div
          class="driver"
        >
          <div
            class="avatar avatar"
          >
            <span>
              ...
            </span>
            <span
              class="number"
            >
              ...
            </span>
          </div>
          <p
            class="name"
          >
            Loading
             
            <b>
              ...
            </b>
          </p>
          <p
            class="car"
          >
            ...
          </p>
        </div>
        <p
          class="pts"
        >
          ...
        </p>
      </div>
    </div>
  </div>
</div>
`;
